### PR TITLE
Provisioner name + provisioner k8s pod definition

### PR DESCRIPTION
**Pull request by aburluka**

### CODE REVIEW

File: vzstorage-pd/provisioner-pod.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  hostNetwork: true
+  containers:
+  - name: virtuozzo-provisioner-pod
+    image: virtuozzo-storage:latest
+    imagePullPolicy: IfNotPresent
+    securityContext:
+      privileged: true
+    volumeMounts:
+      - name: internal
+        mountPath: /var/run/virtuozzo-provisioner
```
Comment (dmission) [inline vzstorage-pd/provisioner-pod.yaml]: /var/run is tmpfs, will be cleared on node restart.

BTW, what if we simply hardcode some identity value? I.e. keep it the same in every vzstorage-provisioner?

**Comment by avagin** [comment]
I've read about "Running multiple provisioners and giving provisioners identities"
https://github.com/kubernetes-incubator/external-storage/tree/master/docs

And now I think the idea to use a file is not so good, because
* we are going to run provisioner in a pod, so each time when kubernetes will re-start a pod, it will have a clean file system, so we will create a new identity. It is what we want to avoid.
* Actually this identity should not be a random number. It may be a node name, if it is bound to a node. It may be a pod name, if it is running in a pod. The idea is that if we run a few provisioner in parallel, each of them has to have uniq name. And if a provisioner is restarted, it has to use the same identity.

I suggest to add an option to specify identity or get it from the environment variable.

**Comment by avagin** [comment]
https://github.com/avagin/external-storage/commit/ad9cdcd76f4df6c6eb77014cee81cc7d4e4df6df

**Comment by aburluka** [comment]
We store this file on some volume, so identity file won't be removed on pod's restart. I've successfully tested pod restart cases

**Comment by avagin** [comment]
@aburluka what volume do you mean? /export is a host path volume, so if a provisioner pod will be stareted on another host, you will have a new identity file.

**Comment by aburluka** [comment]
@avagin User can prepare shared volume and use it in pod definition. Btw, it can be workarounded by using etcd

File: vzstorage-pd/provisioner-pod.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  containers:
+  - name: virtuozzo-provisioner-pod
+    image: virtuozzo-storage:latest
+    imagePullPolicy: IfNotPresent
+    args: ["--name", "vz-provisioner"]
+    securityContext:
+      privileged: true
+    volumeMounts:
+      - name: internal
+        mountPath: /export
```
Comment (dmission) [inline vzstorage-pd/provisioner-pod.yaml]: not needed now

File: vzstorage-pd/virtuozzo-provisioner.go [hunk 3]
```diff
@@ -258,12 +254,12 @@ func (p *vzFSProvisioner) Provision(options controller.VolumeOptions) (*v1.Persi
...
-	ann, ok := volume.Annotations[provisionerIDAnn]
+	ann, ok := volume.Annotations[parentProvisionerAnn]
 	if !ok {
-		return errors.New("identity annotation not found on PV")
+		return errors.New("Parent provisioner name annotation not found on PV")
 	}
-	if ann != string(p.identity) {
-		return &controller.IgnoredError{"identity annotation on PV does not match ours"}
+	if ann != *provisionerId {
+		return &controller.IgnoredError{"parent proviosioner name annotation on PV does not match ours"}
```
Comment (dmission) [inline vzstorage-pd/virtuozzo-provisioner.go]: mistype - proviosioner

**Pull request merged**